src/content/drafts/how-to-create-a-custom-eslint-rule.md

```diff
@@ -0,0 +1,282 @@
+
+Here's what this would look like for the `no-bacon` example:
+
+```javascript
+ruleTester.run("no-bacon", rule, {
+  valid: [
+    "console.log('bacon')",
+    "function ba_con() {}",
+    "class b4c0n {}",
+    "const bac = 'hello'",
+  ],
+
+  invalid: [
+    { code: 'const bacon = "hello"', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'const bacona = "hello"', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'console.log(bacon)', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'console.log(abacon)', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'function bacon() {}', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'function validName(bacon) {}', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'function baConstructor() {}', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'class bacon {}', errors: [{ messageId: "avoidBacon"}] },
+    { code: 'class someBaconClass {}', errors: [{ messageId: "avoidBacon"}] },
+  ],
+});
+```
+
+Right now, you can ignore the `messageId` part of this as the next section will talk about this in more detail. Just know that, when you go to write your rule, you'll need to provide an error message with the `key` of `avoidBacon` (or whatever name fits your rule). 
+
+If you want to see what this file looks like entirely, check out the gist for it [here](https://gist.github.com/mike-solomon/e49714f543a9eac1c6ddca5821195ffb).
+
+With this test file sketched out, you can then run the command `npm test` to execute the tests, but they should all be failing right now.
+
+Also, if you want to learn more about ESLint unit testing and all the options you can configure, check out the [ESLint Rule Unit Tests](https://eslint.org/docs/latest/developer-guide/working-with-rules#rule-unit-tests) doc.
+
+### Source File (lib/rules/no-bacon.js)
+
+This is the file where your core code will be. It contains two main pieces:
+
+* A `meta` object that defines high-level details about your rule such as a description of what it does, what type of rule it is, where you can find the docs, etc.
+* A `create` function that returns an object that ESLint uses to determine whether or not some piece of code violates the rule or not. 
+
+More details about all of the flags and options can be found in the [rule-basics](https://eslint.org/docs/latest/developer-guide/working-with-rules#rule-basics) section of the ESLint docs.
+
+For now, let's briefly talk through each of these pieces. I'll start by showing you some code, and then I'll talk through what each line does (you can find the gist for entire file [here](https://gist.github.com/mike-solomon/1f191393b5bef696937409491c8c559b) if that's easier). 
+
+#### Meta Section
+
+For the bacon example, the `meta` section would look like:
+
+```javascript
+meta: {
```

Comment:
This is technically invalid JS syntax, which may confuse some readers. Can we build this out a bit?

```js
module.exports = {
   meta: {
       ...
    },
   create(context) {
      // omitted for clarity
    }

};
```